app/config.go

```diff
@@ -173,53 +186,57 @@ func loadExternalProperties() (map[string]interface{}, error) {
 			// Override through P1=V1,P2=V2
 			for _, pair := range strings.Split(propFile, ",") {
 				kv := strings.Split(pair, "=")
-				if len(kv) == 2 && kv[0] != "" &&  kv[1] != "" {
+				if len(kv) == 2 && kv[0] != "" && kv[1] != "" {
 					props[kv[0]] = kv[1]
 				} else {
 					logger.Warnf("'%s' is not valid override value. It must be in PropName=PropValue format.", pair)
 				}
 			}
 		}
+	}
 
-		if len(props) > 0 {
-			// Resolve property values
-			resolverType := config.GetAppPropertiesValueResolver()
-			if resolverType != "" {
-				logger.Infof("'%s' is set to '%s'. ", config.ENV_APP_PROPERTY_RESOLVER_KEY, resolverType)
-				resolver := GetPropertyValueResolver(resolverType)
-				if resolver == nil {
-					errMag := fmt.Sprintf("Unsupported resolver type - %s. Resolver not registered.", resolverType)
-					return nil, errors.New(errMag)
-				}
-
-				for k, v := range  props {
-					strVal, ok := v.(string)
-					if ok && len(strVal) > 0 {
-						if strVal[0] == '$' {
-							// Use resolver
-							newVal, err := resolver.ResolveValue(strVal[1:])
-							if err != nil {
-								return nil, err
-							}
-							props[k] = newVal
-
-							// May be a secret??
-							strVal, _ = newVal.(string)
-						}
+	resolverType := config.GetAppPropertiesValueResolver()
+	var resolver PropertyValueResolver
+	if resolverType != "" {
+		logger.Infof("'%s' is set to '%s'. ", config.ENV_APP_PROPERTY_RESOLVER_KEY, resolverType)
+		resolver = GetPropertyValueResolver(resolverType)
+		if resolver == nil {
+			errMag := fmt.Sprintf("Unsupported resolver type - %s. Resolver not registered.", resolverType)
+			return nil, errors.New(errMag)
+		}
+	}
 
-						if len(strVal) > 0 &&  strings.HasPrefix(strVal, "SECRET:") {
-							// Resolve secret value
-							newVal, err := resolveSecretValue(strVal)
-							if err != nil {
-								return nil, err
-							}
+	if resolverType != "" {
```

Comment:
Move those code inside above "if" section, unnecessary to take another if.

Reply:
@lixingwang done in latest update.